meson.build

```diff
@@ -0,0 +1,177 @@
+    server_serialice,
+    server_deserialice,
+]
+
+add_global_arguments([
+    '-Wall',
+    '-Wextra',
+    '-Wno-format-zero-length',
+    '-Wno-unused-parameter',
+    '-Wstrict-prototypes',
+    '-Wmissing-prototypes',
+    '-Wno-missing-field-initializers',
+    '-fvisibility=hidden',
+], language: 'c')
+
+libwaltham_inc = include_directories('src/waltham')
+libwaltham = shared_library('waltham', libwaltham_src, generated_src,
+                            include_directories: libwaltham_inc,
+                            link_args: '-Wl,--no-undefined',
+                            install: true,
+                            version: version,
+                            soversion: version[0])
+install_headers(libwaltham_headers, subdir: 'waltham')
+pkgconfig.generate(libraries: libwaltham,
+                   subdirs: 'waltham',
+                   version: version,
+                   name: 'Waltham',
+                   filebase: 'waltham',
+                   description: desc)
+
+tests_src = [
+    'tests/w-util.c',
+    'tests/w-util.h',
+]
+
+client_src = ['tests/client-api-example.c', client_h, tests_src]
+client_exe = executable('client', client_src, include_directories: libwaltham_inc, link_with: libwaltham)
+test('client', client_exe)
+
+server_src = ['tests/server-api-example.c', server_h, tests_src]
+server_exe = executable('server', server_src, include_directories: libwaltham_inc, link_with: libwaltham)
+test('server', server_exe)
+
+if get_option('documentation')
+    doxygen = find_program('doxygen')
+    sed = find_program('sed')
+
+    doxygen_version_cmd = run_command(doxygen.path(), '--version')
+    if doxygen_version_cmd.returncode() != 0
+        error('Command "doxygen --version" failed.')
+    endif
+    doxygen_version = doxygen_version_cmd.stdout()
+    if doxygen_version.version_compare('< 1.8.0')
+        error('doxygen needs to be at least version 1.8.0 (have @0@)'.format(doxygen_version))
+    endif
+
+    mainpage_md = custom_target('mainpage.md',
+                                input: 'README.md',
+                                output: 'mainpage.md',
+                                command: [sed, '-e', 's|^Waltham IPC Library$|Waltham IPC Library {#mainpage}|', '@INPUT@'],
+                                capture: true)
+
+    doc_src = [
+        mainpage_md,
+        'doc/api_groups.dox',
+        'doc/usage.dox',
+        'src/waltham/waltham-connection.h',
+        'src/waltham/waltham-object.h',
+        'src/waltham/waltham-util.h',
+        client_h,
+        server_h,
+    ]
+
+    doc_config = configuration_data()
+    doc_config.set('PACKAGE_NAME', meson.project_name())
+    doc_config.set('VERSION', version)
+    doc_config.set('top_srcdir', meson.source_root())
+    doc_config.set('top_builddir', meson.build_root())
+
+    doxyfile = configure_file(input: 'doc/doxygen.conf.in',
+                              output: 'doxygen.conf',
+                              configuration: doc_config)
```

Comment:
Forgot to set INPUT in the doxyfile, causing practically all documentation to be missing.

You might also want to look at `commit-gh-pages.bash` script. Its purpose is to commit the already built documentation into the gh-pages branch. It is written assuming one does an in-tree build. We should have another patch to change it to assume an out-of-tree build, probably by assuming it is executed in the builddir. It could check the existence of some file to ensure CWD is the builddir and then just take `doc/html` as it does already.

A nice trait of `commit-gh-pages.bash` is that you can then look at the commit it made to see how the generated documentation has changed.